arcgis-ios-sdk-samples/Features/Feature layer selection/FLSelectionViewController.swift

```diff
@@ -15,78 +15,83 @@
+        return map
     }
     
-    // MARK: - AGSGeoViewTouchDelegate
+    // MARK: UIViewController
     
-    func geoView(_ geoView: AGSGeoView, didTapAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
-        //cancel the active query if it hasn't been completed yet
-        if let activeSelectionQuery = activeSelectionQuery {
-            activeSelectionQuery.cancel()
-        }
-        
-        guard let map = mapView.map,
-            let featureLayer = featureLayer else {
-                return
-        }
-        
-        //tolerance level
-        let toleranceInPoints: Double = 12
-        //use tolerance to compute the envelope for query
-        let toleranceInMapUnits = toleranceInPoints * mapView.unitsPerPoint
-        let envelope = AGSEnvelope(xMin: mapPoint.x - toleranceInMapUnits,
-                                   yMin: mapPoint.y - toleranceInMapUnits,
-                                   xMax: mapPoint.x + toleranceInMapUnits,
-                                   yMax: mapPoint.y + toleranceInMapUnits,
-                                   spatialReference: map.spatialReference)
-        
-        //create query parameters object
-        let queryParams = AGSQueryParameters()
-        queryParams.geometry = envelope
-        
-        //run the selection query
-        activeSelectionQuery = featureLayer.selectFeatures(withQuery: queryParams, mode: .new) { [weak self] (queryResult: AGSFeatureQueryResult?, error: Error?) in
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Add the source code button item to the right of navigation bar.
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = ["FLSelectionViewController"]
+        // Load the feature layer.
+        featureLayer.load { [weak self] (error) in
             if let error = error {
                 self?.presentAlert(error: error)
+            } else {
+                self?.mapView.setViewpointScale(2e8)
             }
-            if let result = queryResult {
-                print("\(result.featureEnumerator().allObjects.count) feature(s) selected")
+        }
+    }
+}
+
+// MARK: - AGSGeoViewTouchDelegate
+
+extension FLSelectionViewController: AGSGeoViewTouchDelegate {
+    func geoView(_ geoView: AGSGeoView, didTapAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        // Identify features at the tapped location.
+        mapView.identifyLayer(featureLayer, screenPoint: screenPoint, tolerance: 12.0, returnPopupsOnly: false, maximumResults: 10) { [weak self] result in
+            guard let self = self else { return }
+            if let error = result.error {
+                self.presentAlert(error: error)
+                return
             }
+            // Un-select previous results.
+            self.featureLayer.unselectFeatures(self.selectedFeatures)
+            // Select current results.
+            self.selectedFeatures = result.geoElements.map { $0 as! AGSFeature }
```

Comment:
No need to use `map(_:)` here. Just cast the array, not each feature individually.

```suggestion
            self.selectedFeatures = result.geoElements as! [AGSFeature]
```